service/internal/graph/nodes.go

```diff
@@ -238,16 +238,13 @@ func (n *connectorNode) buildComponent(
 
 	switch n.rcvrPipelineType {
 	case component.DataTypeTraces:
-		next := nexts[0].(consumer.Traces)
 		capability := nexts[0].Capabilities()
```

Comment:
start with false?

Reply:
@bogdandrutu: Could I ask you to clarify what you're asking for? Previously this code used to assign `next := nexts[0].(consumer.Traces)` and then reassign it to a fanoutconsumer `if len(nexts) > 1`. Now it unconditionally assigns next to a fanoutconsumer. Looking at the original code, it appears that nexts should always have a length of at least 1, is there a case that I am missing?

Reply:
Just do:

```
capability := consumer.Capabilities{MutatesData: false}
```